src/Ocelot/Responder/HttpContextResponder.cs

```diff
@@ -42,18 +42,20 @@ public async Task SetResponseOnHttpContext(HttpContext context, DownstreamRespon
                 AddHeaderIfDoesntExist(context, new Header(httpResponseHeader.Key, httpResponseHeader.Value));
             }
 
-            var content = await response.Content.ReadAsStreamAsync();
-
-            if (response.Content.Headers.ContentLength != null)
+            await using (var content = await response.Content.ReadAsStreamAsync())
             {
-                AddHeaderIfDoesntExist(context, new Header("Content-Length", new[] { response.Content.Headers.ContentLength.ToString() }));
-            }
+                if (response.Content.Headers.ContentLength != null)
+                {
+                    AddHeaderIfDoesntExist(context,
+                        new Header("Content-Length", new[] {response.Content.Headers.ContentLength.ToString()}));
+                }
 
-            await using (content)
-            {
-                if (response.StatusCode != HttpStatusCode.NotModified && context.Response.ContentLength != 0)
+                await using (content)
```

Comment:
Probably the content will be disposed here. But you are right, if anything happens, then it's not disposed. I would recommend moving declaration to line 53

Reply:
Gui, after rebasing commits I saw some changes in develop by you. Seems like you had moved this declaration in your previous PRs (1824).
And when resolving merge conflicts I've chosen the changes from develop (your ones).
Is it right? ))

Reply:
Yes, during the process I checked the disposal, and moved the declaration nearer to usage.
```
if (response.StatusCode != HttpStatusCode.NotModified && context.Response.ContentLength != 0)
        {
            await using var content = await response.Content.ReadAsStreamAsync();
            await content.CopyToAsync(context.Response.Body, context.RequestAborted);
        }
```